src/Summary.jsx

```diff
@@ -64,6 +81,29 @@ class Summary extends React.Component {
     }
   }
 
+  setColor = (name) => {
+    const { colorPerBiome, colors, featuresCounter } = this.state;
+    this.setState({
+      colorPerBiome: [
+        ...colorPerBiome,
+        {
+          name,
+          color: colors[featuresCounter],
+        },
+      ],
+      featuresCounter: featuresCounter + 1,
+    });
+    return colors[featuresCounter];
+  };
+
+  /**
+   * Count biomes to set color
+   */
+  getColorCode = (name) => {
+    const { colorPerBiome } = this.state;
+    return colorPerBiome.find((element) => element.name === name).color;
```

Comment:
It's missing a default case. If you use an object instead of an array, it's easier to find and to return a default case (color).